### PR TITLE
readline: don't use -Wl,--as-needed

### PR DESCRIPTION
**Pull request by pnasrat**
Fixes explicit ncursesw linking

### CODE REVIEW

**Comment by pnasrat** [comment]
cc @imjasonh

**Comment by pnasrat** [comment]
You can see the linking worked when it checks it can be installed and pulls in ncurses

https://github.com/wolfi-dev/os/actions/runs/8913595150/job/24479504501?pr=18158#step:9:116

**Review by imjasonh — APPROVED** [review]

File: readline.yaml
Comment (imjasonh) [inline readline.yaml:1]: Need to increase the epoch to 4 to rebuild with this change

**Review by imjasonh — CHANGES_REQUESTED** [review]

File: readline.yaml
Comment (pnasrat) [reply readline.yaml:1]: Force pushed with epoch bump

**Review by imjasonh — APPROVED** [review]

**Pull request merged**